src/api/app/websocket.ts

```diff
@@ -317,15 +322,15 @@ export class AppWebsocket implements AppClient {
         );
       }
       const cloneCell = appInfo.cell_info[baseRoleName].find(
-        (c) => CellType.Cloned in c && c[CellType.Cloned].clone_id === roleName
+        (c) => c.type === "cloned" && c.value.clone_id === roleName
       );
-      if (!cloneCell || !(CellType.Cloned in cloneCell)) {
+      if (!cloneCell) {
```

Comment:
I much prefer to repeat the condition here than a type assertion.
```suggestion
      if (!cloneCell || cloneCell.type !== CellType.Cloned) {
```

Reply:
Changed. It seems like a weak spot of typescript here that it doesn't know that an array filtered by type === "cloned" can only contain cells of type cloned, hence why I thought that check is redundant. But probably still worth to be explicit in checking.

Reply:
It is definitely lame.